x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/pages/coverage_overview/reducer.ts

```diff
@@ -5,27 +5,75 @@
  * 2.0.
  */
 
-export interface State {
+import type {
+  CoverageOverviewFilter,
+  CoverageOverviewRuleActivity,
+  CoverageOverviewRuleSource,
+} from '../../../../../common/api/detection_engine';
+
+export interface CoverageOverviewDashboardState {
   showExpandedCells: boolean;
+  filter: CoverageOverviewFilter;
+}
+
+export interface CoverageOverviewDashboardContextType {
+  state: CoverageOverviewDashboardState;
+  dispatch: React.Dispatch<Action>;
 }
 
-export const initialState: State = {
+export const initialState: CoverageOverviewDashboardState = {
   showExpandedCells: false,
+  filter: {},
 };
 
-export interface Action {
-  type: 'setShowExpandedCells';
-  value: boolean;
-}
+export type Action =
+  | {
+      type: 'setShowExpandedCells';
+      value: boolean;
+    }
+  | {
+      type: 'setRuleStatusFilter';
+      value: CoverageOverviewRuleActivity[];
+    }
+  | {
+      type: 'setRuleTypeFilter';
+      value: CoverageOverviewRuleSource[];
+    }
+  | {
+      type: 'setRuleSearchFilter';
+      value: string;
+    };
 
 export const createCoverageOverviewDashboardReducer =
   () =>
-  (state: State, action: Action): State => {
+  (state: CoverageOverviewDashboardState, action: Action): CoverageOverviewDashboardState => {
     switch (action.type) {
       case 'setShowExpandedCells': {
         const { value } = action;
         return { ...state, showExpandedCells: value };
       }
+      case 'setRuleStatusFilter': {
+        const { value } = action;
+        if (value.length === 0 || value.length === 2) {
```

Comment:
@maximpn This weird hardcoded edge case with the `length === 2` is caused by the api not taking both `Enabled` and `Disabled` as filters (because of the way the ternary written is structured [here](https://github.com/elastic/kibana/blob/24655504af205f4a5406c19ce4b36ee6a82d17a6/x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/api/rules/coverage_overview/handle_coverage_overview_request.ts#L41)), even though we seem to have the other rule filter be an `OR` instead of an `AND`. Is this the expected action?

Reply:
As empty activity filter produces the same result as as all selected activity values it makes sense to omit the filter param in the request when all values are selected. I don't think there should be an ability to do the same thing in two ways. But you are right the API endpoint doesn't handle right requests when activity is set to `['enabled', 'disabled']`. I thought to change it during milestone 2 but it makes sense to do it earlier.